src/custom/utils/metadata.ts

```diff
@@ -49,3 +59,17 @@ export function generateAppDataDoc(metadata: MetadataDoc = {}): AppDataDoc {
     },
   }
 }
+
+export async function uploadMetadataDocToIpfs(appDataDoc: AppDataDoc): Promise<string> {
+  try {
+    const client = create({ url: 'https://ipfs.infura.io:5001/api/v0' })
+    const doc = JSON.stringify(appDataDoc)
```

Comment:
Probably doesn't matter much, but since it has been implemented already, could you use `safeStringify.stableStringify(value)` instead: see https://github.com/gnosis/cowswap/blob/866181b4af7fd705d56e7597d2191692075d1204/src/custom/utils/deterministicHash.ts#L6